### PR TITLE
Add aria-label to the toggle sidebar button

### PR DESCRIPTION
**Pull request by diegohaz**
I'd like to get familiar with the codebase, and I found that this button didn't have a label. So I added it.

There're other accessibility issues with this component, like the lack of `aria-expanded`, `aria-haspoup` and `aria-controls` attributes and the lack of focus management (focus should probably go into the settings sidebar, so users don't get lost when clicking on this button). I can open a separate issue for these and address them once I get more used to the codebase.

#### Changes proposed in this Pull Request

* Add `aria-label` to the toggle post settings button

#### Testing instructions

* Click on `Add new post`
* Inspect the button with a gear icon

#### Before

image

#### After

image

image

### CODE REVIEW

File: client/post-editor/editor-ground-control/index.jsx [hunk 1]
```diff
@@ -143,7 +149,7 @@ export class EditorGroundControl extends React.Component {
 	}
 
 	getCloseButtonPath() {
-		const editorPathRegex = /^(\/block-editor)?\/(post|page|(edit\/[^\/]+))(\/|$)/i;
+		const editorPathRegex = /^(\/block-editor)?\/(post|page|(edit\/[^/]+))(\/|$)/i;
```
Comment (obenland) [inline client/post-editor/editor-ground-control/index.jsx]: Is this change related to the label change?

Comment (diegohaz) [reply client/post-editor/editor-ground-control/index.jsx]: Not at all. It was just the linter yelling at me (pre-commit).

**Comment by lancewillett** [comment]
Closing this one. Though a good change, the editor UI in question is now deprecated.
